### PR TITLE
Make AutoMM dataloader return feature column information

### PR DESCRIPTION
**Pull request by zhiqiangdon**
To do feature column level distillation, we need to let AutoMM dataloader return feature column information.

1. Add feature column names in data processors.
2. Dataframe preprocessor outputs dictionary instead of list.
3. Data processors's return contains feature column information.
4. Use property for return keys in data processors.
5. CLIP uses eos token as the cls token. So avoid using eos as the sep token for CLIP.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1710-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1710/2/index.html

File: text/src/autogluon/text/automm/data/process_categorical.py [hunk 1]
```diff
@@ -57,16 +75,20 @@ def process_one_sample(
         -------
         A dictionary containing the processed categorical features.
         """
-        return {
-            f"{self.prefix}_{CATEGORICAL}": categorical_features
-        }
+        ret = {}
+        for i, col_name in enumerate(categorical_features.keys()):
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/data/process_categorical.py:79]: You may add a TODO item about the potential performance issue.

Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/data/process_categorical.py:80]: Done. We may move the for loop into __init__(), but need to consider the backward compatibility.

**Review by bryanyzhu — APPROVED** [review]

**Review by sxjscience — APPROVED** [review]

**Comment by szha** [comment]
Job PR-1710-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1710/3/index.html

**Pull request merged**